mne_connectivity/effective.py

```diff
@@ -243,3 +247,191 @@ def phase_slope_index(
+    give an indication of the direction of the information flow (causality).
+    For two time series, and one computes the PSI between the first and the
+    second time series as follows
+
+    indices = (np.array([0]), np.array([1]))
+    psi = phase_slope_index(data, indices=indices, ...)
+
+    A positive value means that time series 0 is ahead of time series 1 and
+    a negative value means the opposite.
+
+    The PSI is computed from the coherency (see spectral_connectivity_epochs),
+    details can be found in :footcite:`NolteEtAl2008`.
+
+    This function computes PSI over time from epoched data.
+    The data may consist of a single epoch.
+
+    Parameters
+    ----------
+    data : array-like, shape=(n_epochs, n_signals, n_times)
+        Can also be a list/generator of array, shape =(n_signals, n_times);
+        list/generator of SourceEstimate; or Epochs.
+        The data from which to compute connectivity. Note that it is also
+        possible to combine multiple signals by providing a list of tuples,
+        e.g., data = [(arr_0, stc_0), (arr_1, stc_1), (arr_2, stc_2)],
+        corresponds to 3 epochs, and arr_* could be an array with the same
+        number of time points as stc_*.
+    %(names)s
+    indices : tuple of array | None
+        Two arrays with indices of connections for which to compute
+        connectivity. If None, all connections are computed.
+    sfreq : float
+        The sampling frequency.
+    mode : str
+        Spectrum estimation mode can be either: 'multitaper', 'fourier', or
+        'cwt_morlet'.
+    fmin : float | tuple of float
+        The lower frequency of interest. Multiple bands are defined using
+        a tuple, e.g., (8., 20.) for two bands with 8Hz and 20Hz lower freq.
+        If None the frequency corresponding to an epoch length of 5 cycles
+        is used.
+    fmax : float | tuple of float
+        The upper frequency of interest. Multiple bands are dedined using
+        a tuple, e.g. (13., 30.) for two band with 13Hz and 30Hz upper freq.
+    mt_bandwidth : float | None
+        The bandwidth of the multitaper windowing function in Hz.
+        Only used in 'multitaper' mode.
+    freqs : array
+        Array of frequencies of interest. Only used in 'cwt_morlet' mode.
+    n_cycles : float | array of float
+        Number of cycles. Fixed number or one per frequency. Only used in
+        'cwt_morlet' mode.
+    n_jobs : int
+        How many epochs to process in parallel.
+    %(verbose)s
+
+    Returns
+    -------
+    conn : instance of Connectivity
+        Computed connectivity measure(s). ``EpochSpectralConnectivity``
+        container. The shape of each array is
+        (n_signals ** 2, n_bands, n_epochs)
+        when "indices" is None, or
+        (n_con, n_bands, n_epochs) 
+        when "indices" is specified and "n_con = len(indices[0])".
+
+    See Also
+    --------
+    mne_connectivity.EpochSpectralConnectivity
```

Comment:
I'm wondering if it's good to reiterate that `spectral_connectivity_time()` is a relevant function. Though I realise you based this on the existing function docstring which doesn't link to `spec_conn_epochs()`.

Reply:
Yes, I think this is a good idea.